src/Domain/DoctrineDonationRepository.php

```diff
@@ -741,6 +741,33 @@ public function findAllCompleteForCustomer(StripeCustomerId $stripeCustomerId):
         return $result;
     }
 
+    /**
+     * We only set Payment Intent on the day of the payment due to stripe limitations, see
+     */
+    public function findDonationsToSetPaymentIntent(\DateTimeImmutable $atDateTime, int $limit): array
+    {
+        $preAuthorized = DonationStatus::PreAuthorized->value;
+        $active = MandateStatus::Active->value;
+        $currentDateTime = new \DateTimeImmutable('now');
```

Comment:
This fn is using a mixture of a DI'd datetime and 'now' which is going to make it hard to test and confusing to use. We should probably use `$atDateTime` consistently?